comp/otelcol/configstore/impl/configstore.go

```diff
@@ -26,18 +28,44 @@ func NewConfigStore() (configstore.Component, error) {
 	return &configStoreImpl{}, nil
 }
 
-// AddProvidedConf stores the config into configStoreImpl.
-func (c *configStoreImpl) AddProvidedConf(config *otelcol.Config) {
-	c.mu.RLock()
-	defer c.mu.RUnlock()
+func (c *configStoreImpl) AddConfigs(providedCPS otelcol.ConfigProviderSettings, enhancedCPS otelcol.ConfigProviderSettings, factories otelcol.Factories) error {
```

Comment:
Can we add them as Fx deps. It seems odd to post construction

Reply:
Not sure to understand what you mean here ?

Reply:
On further thought, we can ignore this for now. We will revisit this later